File: docs/src/testers/sessions/default_proxy.ts
// @sniptest filename=default_proxy.ts
// @sniptest show=1-11
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

const session = client.Session({
  proxies: true,
});
await session.use(async () => {
  await session.execute({ type: 'goto', url: 'https://www.notte.cc/' });
  await session.observe();
});

const status = await session.status();
if (status.proxies !== true) throw new Error('Session proxies mismatch');

export { status };

File: docs/src/testers/sessions/default_proxy.py
# @sniptest filename=default_proxy.py
# @sniptest show=1-8
from notte_sdk import NotteClient

client = NotteClient()

# Start a session with built-in proxies
with client.Session(proxies=True) as session:
    _ = session.execute(type="goto", url="https://www.notte.cc/")
    _ = session.observe()

status = session.status()
assert status.proxies is True
